Makefile

```diff
@@ -25,7 +25,7 @@ clean:
 
 .PHONY: deps
 deps: $(VENV)
-	$(PIP) install -r requirements.txt
+	$(PIP) install .[dev]
```

Comment:
`make test` looks safe as well, but may be worth a test

Reply:
Seems to be working as expected.

<details>

<summary> make test console output

```bash

(detection_dev) ➜  detection-rules git:(create-setup-cfg) ✗ make test
pip3 install virtualenv
Looking in indexes: https://pypi.org/simple, https://artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Collecting virtualenv
  Downloading virtualenv-20.16.7-py3-none-any.whl (8.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.8/8.8 MB 3.2 MB/s eta 0:00:00
Collecting distlib<1,>=0.3.6
  Using cached distlib-0.3.6-py2.py3-none-any.whl (468 kB)
Collecting filelock<4,>=3.4.1
  Using cached filelock-3.8.0-py3-none-any.whl (10 kB)
Requirement already satisfied: platformdirs<3,>=2.4 in /Users/stryker/.virtualenvs/detection_dev/lib/python3.10/site-packages (from virtualenv) (2.5.2)
Installing collected packages: distlib, filelock, virtualenv
Successfully installed distlib-0.3.6 filelock-3.8.0 virtualenv-20.16.7
WARNING: You are using pip version 22.0.4; however, version 22.3.1 is available.
You should consider upgrading via the '/Users/stryker/.virtualenvs/detection_dev/bin/python -m pip install --upgrade pip' command.
virtualenv ./env/detection-rules-build --python=python3.8
RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.8'
make: *** [env/detection-rules-build] Error 1
(detection_dev) ➜  detection-rules git:(create-setup-cfg) ✗ make test
pip3 install virtualenv
Looking in indexes: https://pypi.org/simple, https://artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Requirement already satisfied: virtualenv in /Users/stryker/.virtualenvs/detection_dev/lib/python3.10/site-packages (20.16.7)
Requirement already satisfied: distlib<1,>=0.3.6 in /Users/stryker/.virtualenvs/detection_dev/lib/python3.10/site-packages (from virtualenv) (0.3.6)
Requirement already satisfied: platformdirs<3,>=2.4 in /Users/stryker/.virtualenvs/detection_dev/lib/python3.10/site-packages (from virtualenv) (2.5.2)
Requirement already satisfied: filelock<4,>=3.4.1 in /Users/stryker/.virtualenvs/detection_dev/lib/python3.10/site-packages (from virtualenv) (3.8.0)
WARNING: You are using pip version 22.0.4; however, version 22.3.1 is available.
You should consider upgrading via the '/Users/stryker/.virtualenvs/detection_dev/bin/python -m pip install --upgrade pip' command.
virtualenv ./env/detection-rules-build --python=python3
created virtual environment CPython3.10.8.final.0-64 in 372ms
  creator CPython3Posix(dest=/Users/stryker/workspace/Elastic/detection-rules/env/detection-rules-build, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/stryker/Library/Application Support/virtualenv)
    added seed packages: pip==22.3, setuptools==65.5.0, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
./env/detection-rules-build/bin/python -m pip install .[dev]
Looking in indexes: https://pypi.org/simple, https://artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Processing /Users/stryker/workspace/Elastic/detection-rules
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting marshmallow~=3.13.0
  Using cached marshmallow-3.13.0-py2.py3-none-any.whl (47 kB)
Collecting PyYAML~=5.3
  Using cached PyYAML-5.4.1.tar.gz (175 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting pytoml
  Using cached pytoml-0.1.21-py2.py3-none-any.whl (8.5 kB)
Collecting Click~=8.1.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting marko
  Downloading marko-1.2.2-py3-none-any.whl (39 kB)
Collecting typing-inspect==0.7.1
  Using cached typing_inspect-0.7.1-py3-none-any.whl (8.4 kB)
Collecting toml==0.10.0
  Using cached toml-0.10.0-py2.py3-none-any.whl (25 kB)
Collecting marshmallow-jsonschema~=0.12.0
  Using cached marshmallow_jsonschema-0.12.0-py3-none-any.whl (11 kB)
Collecting marshmallow-dataclass[union]~=8.5.6
  Downloading marshmallow_dataclass-8.5.10-py3-none-any.whl (17 kB)
Collecting XlsxWriter~=1.3.6
  Using cached XlsxWriter-1.3.9-py2.py3-none-any.whl (145 kB)
Collecting elasticsearch~=8.1
  Downloading elasticsearch-8.5.0-py3-none-any.whl (385 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 385.3/385.3 kB 1.7 MB/s eta 0:00:00
Collecting jsl==0.2.4
  Using cached jsl-0.2.4-py3-none-any.whl
Collecting requests~=2.27
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting jsonschema==3.2.0
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting eql==0.9.15
  Using cached eql-0.9.15-py2.py3-none-any.whl (105 kB)
Collecting marshmallow-union~=0.1.15
  Using cached marshmallow_union-0.1.15.post1-py2.py3-none-any.whl (4.6 kB)
Collecting flake8==3.8.1
  Using cached flake8-3.8.1-py2.py3-none-any.whl (72 kB)
Collecting PyGithub==1.55
  Using cached PyGithub-1.55-py3-none-any.whl (291 kB)
Collecting pytest>=3.6
  Using cached pytest-7.2.0-py3-none-any.whl (316 kB)
Collecting pep8-naming==0.7.0
  Using cached pep8_naming-0.7.0-py2.py3-none-any.whl (7.7 kB)
Collecting pyflakes==2.2.0
  Using cached pyflakes-2.2.0-py2.py3-none-any.whl (66 kB)
Collecting lark-parser~=0.12.0
  Using cached lark_parser-0.12.0-py2.py3-none-any.whl (103 kB)
Collecting mccabe<0.7.0,>=0.6.0
  Using cached mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Collecting pycodestyle<2.7.0,>=2.6.0a1
  Using cached pycodestyle-2.6.0-py2.py3-none-any.whl (41 kB)
Collecting pyrsistent>=0.14.0
  Using cached pyrsistent-0.19.2-cp310-cp310-macosx_10_9_universal2.whl (82 kB)
Requirement already satisfied: setuptools in ./env/detection-rules-build/lib/python3.10/site-packages (from jsonschema==3.2.0->detection-rules==8.6) (65.5.0)
Collecting six>=1.11.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting attrs>=17.4.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting flake8-polyfill<2,>=1.0.2
  Using cached flake8_polyfill-1.0.2-py2.py3-none-any.whl (7.3 kB)
Collecting pyjwt>=2.0
  Downloading PyJWT-2.6.0-py3-none-any.whl (20 kB)
Collecting deprecated
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting pynacl>=1.4.0
  Using cached PyNaCl-1.5.0-cp36-abi3-macosx_10_10_universal2.whl (349 kB)
Collecting mypy-extensions>=0.3.0
  Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Collecting typing-extensions>=3.7.4
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting elastic-transport<9,>=8
  Using cached elastic_transport-8.4.0-py3-none-any.whl (59 kB)
Collecting typeguard
  Using cached typeguard-2.13.3-py3-none-any.whl (17 kB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting tomli>=1.0.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting exceptiongroup>=1.0.0rc8
  Downloading exceptiongroup-1.0.2-py3-none-any.whl (12 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting cffi>=1.4.1
  Using cached cffi-1.15.1-cp310-cp310-macosx_11_0_arm64.whl (174 kB)
Collecting wrapt<2,>=1.10
  Using cached wrapt-1.14.1-cp310-cp310-macosx_11_0_arm64.whl (35 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Building wheels for collected packages: PyYAML, detection-rules
  Building wheel for PyYAML (pyproject.toml) ... done
  Created wheel for PyYAML: filename=PyYAML-5.4.1-cp310-cp310-macosx_12_0_arm64.whl size=45660 sha256=d0516da9d1fa37b574c9e063d852ef414eded3d66282f912e5413fbbf8ecd8d6
  Stored in directory: /Users/stryker/Library/Caches/pip/wheels/10/3f/c1/6feec52270653969575b83de934b04be56d7a9e961be5041ea
  Building wheel for detection-rules (pyproject.toml) ... done
  Created wheel for detection-rules: filename=detection_rules-8.6-py3-none-any.whl size=19354441 sha256=13cd099fbb55569aae2ca24bd347a7668aae11129b9df10e1b74992c53c05b28
  Stored in directory: /Users/stryker/Library/Caches/pip/wheels/e4/1f/19/1c6547e411e45f8b31b8ca11924b08c5831f7452bf4b038008
Successfully built PyYAML detection-rules
Installing collected packages: XlsxWriter, toml, pytoml, mypy-extensions, mccabe, lark-parser, jsl, iniconfig, wrapt, urllib3, typing-extensions, typeguard, tomli, six, PyYAML, pyrsistent, pyparsing, pyjwt, pyflakes, pycparser, pycodestyle, pluggy, marshmallow, marko, idna, exceptiongroup, eql, Click, charset-normalizer, certifi, attrs, typing-inspect, requests, packaging, marshmallow-union, marshmallow-jsonschema, jsonschema, flake8, elastic-transport, deprecated, cffi, pytest, pynacl, marshmallow-dataclass, flake8-polyfill, elasticsearch, PyGithub, pep8-naming, detection-rules
Successfully installed Click-8.1.3 PyGithub-1.55 PyYAML-5.4.1 XlsxWriter-1.3.9 attrs-22.1.0 certifi-2022.9.24 cffi-1.15.1 charset-normalizer-2.1.1 deprecated-1.2.13 detection-rules-8.6 elastic-transport-8.4.0 elasticsearch-8.5.0 eql-0.9.15 exceptiongroup-1.0.2 flake8-3.8.1 flake8-polyfill-1.0.2 idna-3.4 iniconfig-1.1.1 jsl-0.2.4 jsonschema-3.2.0 lark-parser-0.12.0 marko-1.2.2 marshmallow-3.13.0 marshmallow-dataclass-8.5.10 marshmallow-jsonschema-0.12.0 marshmallow-union-0.1.15.post1 mccabe-0.6.1 mypy-extensions-0.4.3 packaging-21.3 pep8-naming-0.7.0 pluggy-1.0.0 pycodestyle-2.6.0 pycparser-2.21 pyflakes-2.2.0 pyjwt-2.6.0 pynacl-1.5.0 pyparsing-3.0.9 pyrsistent-0.19.2 pytest-7.2.0 pytoml-0.1.21 requests-2.28.1 six-1.16.0 toml-0.10.0 tomli-2.0.1 typeguard-2.13.3 typing-extensions-4.4.0 typing-inspect-0.7.1 urllib3-1.26.12 wrapt-1.14.1

[notice] A new release of pip available: 22.3 -> 22.3.1
[notice] To update, run: python -m pip install --upgrade pip
./env/detection-rules-build/bin/python -m pip install setuptools -U
Looking in indexes: https://pypi.org/simple, https://artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Requirement already satisfied: setuptools in ./env/detection-rules-build/lib/python3.10/site-packages (65.5.0)
Collecting setuptools
  Using cached setuptools-65.5.1-py3-none-any.whl (1.2 MB)
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 65.5.0
    Uninstalling setuptools-65.5.0:
      Successfully uninstalled setuptools-65.5.0
Successfully installed setuptools-65.5.1

[notice] A new release of pip available: 22.3 -> 22.3.1
[notice] To update, run: python -m pip install --upgrade pip
./env/detection-rules-build/bin/python -m pip install .[dev]
Looking in indexes: https://pypi.org/simple, https://artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Processing /Users/stryker/workspace/Elastic/detection-rules
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: marko in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (1.2.2)
Requirement already satisfied: pytoml in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.1.21)
Requirement already satisfied: typing-inspect==0.7.1 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.7.1)
Requirement already satisfied: marshmallow-jsonschema~=0.12.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.12.0)
Requirement already satisfied: PyYAML~=5.3 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (5.4.1)
Requirement already satisfied: toml==0.10.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.10.0)
Requirement already satisfied: Click~=8.1.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (8.1.3)
Requirement already satisfied: requests~=2.27 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (2.28.1)
Requirement already satisfied: marshmallow-dataclass[union]~=8.5.6 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (8.5.10)
Requirement already satisfied: XlsxWriter~=1.3.6 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (1.3.9)
Requirement already satisfied: elasticsearch~=8.1 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (8.5.0)
Requirement already satisfied: marshmallow-union~=0.1.15 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.1.15.post1)
Requirement already satisfied: marshmallow~=3.13.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (3.13.0)
Requirement already satisfied: jsonschema==3.2.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (3.2.0)
Requirement already satisfied: jsl==0.2.4 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.2.4)
Requirement already satisfied: eql==0.9.15 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.9.15)
Requirement already satisfied: pep8-naming==0.7.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (0.7.0)
Requirement already satisfied: pyflakes==2.2.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (2.2.0)
Requirement already satisfied: PyGithub==1.55 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (1.55)
Requirement already satisfied: flake8==3.8.1 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (3.8.1)
Requirement already satisfied: pytest>=3.6 in ./env/detection-rules-build/lib/python3.10/site-packages (from detection-rules==8.6) (7.2.0)
Requirement already satisfied: lark-parser~=0.12.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from eql==0.9.15->detection-rules==8.6) (0.12.0)
Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from flake8==3.8.1->detection-rules==8.6) (0.6.1)
Requirement already satisfied: pycodestyle<2.7.0,>=2.6.0a1 in ./env/detection-rules-build/lib/python3.10/site-packages (from flake8==3.8.1->detection-rules==8.6) (2.6.0)
Requirement already satisfied: pyrsistent>=0.14.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from jsonschema==3.2.0->detection-rules==8.6) (0.19.2)
Requirement already satisfied: attrs>=17.4.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from jsonschema==3.2.0->detection-rules==8.6) (22.1.0)
Requirement already satisfied: setuptools in ./env/detection-rules-build/lib/python3.10/site-packages (from jsonschema==3.2.0->detection-rules==8.6) (65.5.1)
Requirement already satisfied: six>=1.11.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from jsonschema==3.2.0->detection-rules==8.6) (1.16.0)
Requirement already satisfied: flake8-polyfill<2,>=1.0.2 in ./env/detection-rules-build/lib/python3.10/site-packages (from pep8-naming==0.7.0->detection-rules==8.6) (1.0.2)
Requirement already satisfied: pyjwt>=2.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from PyGithub==1.55->detection-rules==8.6) (2.6.0)
Requirement already satisfied: pynacl>=1.4.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from PyGithub==1.55->detection-rules==8.6) (1.5.0)
Requirement already satisfied: deprecated in ./env/detection-rules-build/lib/python3.10/site-packages (from PyGithub==1.55->detection-rules==8.6) (1.2.13)
Requirement already satisfied: typing-extensions>=3.7.4 in ./env/detection-rules-build/lib/python3.10/site-packages (from typing-inspect==0.7.1->detection-rules==8.6) (4.4.0)
Requirement already satisfied: mypy-extensions>=0.3.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from typing-inspect==0.7.1->detection-rules==8.6) (0.4.3)
Requirement already satisfied: elastic-transport<9,>=8 in ./env/detection-rules-build/lib/python3.10/site-packages (from elasticsearch~=8.1->detection-rules==8.6) (8.4.0)
Requirement already satisfied: typeguard in ./env/detection-rules-build/lib/python3.10/site-packages (from marshmallow-dataclass[union]~=8.5.6->detection-rules==8.6) (2.13.3)
Requirement already satisfied: tomli>=1.0.0 in ./env/detection-rules-build/lib/python3.10/site-packages (from pytest>=3.6->detection-rules==8.6) (2.0.1)
Requirement already satisfied: pluggy<2.0,>=0.12 in ./env/detection-rules-build/lib/python3.10/site-packages (from pytest>=3.6->detection-rules==8.6) (1.0.0)
Requirement already satisfied: exceptiongroup>=1.0.0rc8 in ./env/detection-rules-build/lib/python3.10/site-packages (from pytest>=3.6->detection-rules==8.6) (1.0.2)
Requirement already satisfied: packaging in ./env/detection-rules-build/lib/python3.10/site-packages (from pytest>=3.6->detection-rules==8.6) (21.3)
Requirement already satisfied: iniconfig in ./env/detection-rules-build/lib/python3.10/site-packages (from pytest>=3.6->detection-rules==8.6) (1.1.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./env/detection-rules-build/lib/python3.10/site-packages (from requests~=2.27->detection-rules==8.6) (1.26.12)
Requirement already satisfied: charset-normalizer<3,>=2 in ./env/detection-rules-build/lib/python3.10/site-packages (from requests~=2.27->detection-rules==8.6) (2.1.1)
Requirement already satisfied: idna<4,>=2.5 in ./env/detection-rules-build/lib/python3.10/site-packages (from requests~=2.27->detection-rules==8.6) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in ./env/detection-rules-build/lib/python3.10/site-packages (from requests~=2.27->detection-rules==8.6) (2022.9.24)
Requirement already satisfied: cffi>=1.4.1 in ./env/detection-rules-build/lib/python3.10/site-packages (from pynacl>=1.4.0->PyGithub==1.55->detection-rules==8.6) (1.15.1)
Requirement already satisfied: wrapt<2,>=1.10 in ./env/detection-rules-build/lib/python3.10/site-packages (from deprecated->PyGithub==1.55->detection-rules==8.6) (1.14.1)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in ./env/detection-rules-build/lib/python3.10/site-packages (from packaging->pytest>=3.6->detection-rules==8.6) (3.0.9)
Requirement already satisfied: pycparser in ./env/detection-rules-build/lib/python3.10/site-packages (from cffi>=1.4.1->pynacl>=1.4.0->PyGithub==1.55->detection-rules==8.6) (2.21)
Building wheels for collected packages: detection-rules
  Building wheel for detection-rules (pyproject.toml) ... done
  Created wheel for detection-rules: filename=detection_rules-8.6-py3-none-any.whl size=19354441 sha256=feb3477c2a0ada5117767b0090a8392596c0c53677c24feca207334590fdd4d9
  Stored in directory: /Users/stryker/Library/Caches/pip/wheels/e4/1f/19/1c6547e411e45f8b31b8ca11924b08c5831f7452bf4b038008
Successfully built detection-rules
Installing collected packages: detection-rules
  Attempting uninstall: detection-rules
    Found existing installation: detection-rules 8.6
    Uninstalling detection-rules-8.6:
      Successfully uninstalled detection-rules-8.6
Successfully installed detection-rules-8.6

[notice] A new release of pip available: 22.3 -> 22.3.1
[notice] To update, run: python -m pip install --upgrade pip
LINTING
./env/detection-rules-build/bin/python -m flake8 tests detection_rules --ignore D203 --max-line-length 120
./env/detection-rules-build/bin/python -m detection_rules test
Loaded config file: /Users/stryker/workspace/Elastic/detection-rules/.detection-rules-cfg.json

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

======================================================================================================================================================================================================================================================================= test session starts =======================================================================================================================================================================================================================================================================
platform darwin -- Python 3.10.8, pytest-7.2.0, pluggy-1.0.0 -- /Users/stryker/workspace/Elastic/detection-rules/env/detection-rules-build/bin/python
cachedir: .pytest_cache
rootdir: /Users/stryker/workspace/Elastic/detection-rules
plugins: typeguard-2.13.3
collected 116 items

tests/test_all_rules.py::TestValidRules::test_all_rule_queries_optimized PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [  0%]
tests/test_all_rules.py::TestValidRules::test_duplicate_file_names PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [  1%]
tests/test_all_rules.py::TestValidRules::test_file_names PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [  2%]
tests/test_all_rules.py::TestValidRules::test_production_rules_have_rta PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [  3%]
tests/test_all_rules.py::TestValidRules::test_rule_type_changes PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [  4%]
tests/test_all_rules.py::TestValidRules::test_schema_and_dupes PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [  5%]
tests/test_all_rules.py::TestThreatMappings::test_duplicated_tactics PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 [  6%]
tests/test_all_rules.py::TestThreatMappings::test_tactic_to_technique_correlations PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [  6%]
tests/test_all_rules.py::TestThreatMappings::test_technique_deprecations PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [  7%]
tests/test_all_rules.py::TestRuleTags::test_casing_and_spacing PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [  8%]
tests/test_all_rules.py::TestRuleTags::test_primary_tactic_as_tag PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    [  9%]
tests/test_all_rules.py::TestRuleTags::test_required_tags PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 10%]
tests/test_all_rules.py::TestRuleTimelines::test_timeline_has_title PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  [ 11%]
tests/test_all_rules.py::TestRuleFiles::test_rule_file_name_tactic PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [ 12%]
tests/test_all_rules.py::TestRuleMetadata::test_all_min_stack_rules_have_comment PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 12%]
tests/test_all_rules.py::TestRuleMetadata::test_deprecated_rules PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 13%]
tests/test_all_rules.py::TestRuleMetadata::test_integration PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 14%]
tests/test_all_rules.py::TestRuleMetadata::test_rule_demotions PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 15%]
tests/test_all_rules.py::TestRuleMetadata::test_updated_date_newer_than_creation PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 16%]
tests/test_all_rules.py::TestRuleTiming::test_eql_interval_to_maxspan PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 17%]
tests/test_all_rules.py::TestRuleTiming::test_eql_lookback PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 18%]
tests/test_all_rules.py::TestRuleTiming::test_event_override PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         [ 18%]
tests/test_all_rules.py::TestRuleTiming::test_required_lookback PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [ 19%]
tests/test_all_rules.py::TestLicense::test_elastic_license_only_v2 PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [ 20%]
tests/test_all_rules.py::TestIntegrationRules::test_integration_guide PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 21%]
tests/test_all_rules.py::TestIncompatibleFields::test_rule_backports_for_restricted_fields PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 22%]
tests/test_all_rules.py::TestBuildTimeFields::test_build_fields_min_stack PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 23%]
tests/test_all_rules.py::TestRiskScoreMismatch::test_rule_risk_score_severity_mismatch PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                               [ 24%]
tests/test_gh_workflows.py::TestWorkflows::test_matrix_to_lock_version_defaults PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [ 25%]
tests/test_mappings.py::TestMappings::test_false_positives PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 25%]
tests/test_mappings.py::TestMappings::test_true_positives PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 26%]
tests/test_mappings.py::TestRTAs::test_rtas_with_triggered_rules_have_uuid PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 27%]
tests/test_packages.py::TestPackages::test_package_loader_default_configs PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 28%]
tests/test_packages.py::TestPackages::test_package_loader_production_config PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 29%]
tests/test_packages.py::TestPackages::test_package_summary PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 30%]
tests/test_packages.py::TestPackages::test_rule_versioning PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 31%]
tests/test_packages.py::TestRegistryPackage::test_registry_package_config PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 31%]
tests/test_schemas.py::TestSchemas::test_eql_validation PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 32%]
tests/test_schemas.py::TestSchemas::test_query_downgrade_7_x PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         [ 33%]
tests/test_schemas.py::TestSchemas::test_query_downgrade_8_x PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         [ 34%]
tests/test_schemas.py::TestSchemas::test_threshold_downgrade_7_x PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 35%]
tests/test_schemas.py::TestSchemas::test_threshold_downgrade_8_x PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 36%]
tests/test_schemas.py::TestSchemas::test_versioned_downgrade_7_x PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 37%]
tests/test_schemas.py::TestSchemas::test_versioned_downgrade_8_x PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 37%]
tests/test_schemas.py::TestVersionLockSchema::test_version_lock_has_nested_previous PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                  [ 38%]
tests/test_schemas.py::TestVersionLockSchema::test_version_lock_no_previous PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 39%]
tests/test_schemas.py::TestVersions::test_stack_schema_map PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 40%]
tests/test_toml_formatter.py::TestRuleTomlFormatter::test_formatter_deep PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 41%]
tests/test_toml_formatter.py::TestRuleTomlFormatter::test_formatter_rule PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 42%]
tests/test_toml_formatter.py::TestRuleTomlFormatter::test_normalization PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 43%]
tests/test_utils.py::TestTimeUtils::test_caching PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 43%]
tests/test_utils.py::TestTimeUtils::test_event_class_normalization PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [ 44%]
tests/test_utils.py::TestTimeUtils::test_schema_multifields PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 45%]
tests/test_utils.py::TestTimeUtils::test_time_normalize PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 46%]
tests/test_version_locking.py::TestVersionLock::test_previous_entries_gte_current_min_stack PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 47%]
tests/kuery/test_dsl.py::TestKQLtoDSL::test_and_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 48%]
tests/kuery/test_dsl.py::TestKQLtoDSL::test_field_exists PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 49%]
tests/kuery/test_dsl.py::TestKQLtoDSL::test_field_inequality PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         [ 50%]
tests/kuery/test_dsl.py::TestKQLtoDSL::test_field_match PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 50%]
tests/kuery/test_dsl.py::TestKQLtoDSL::test_not_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 51%]
tests/kuery/test_dsl.py::TestKQLtoDSL::test_optimizations PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 52%]
tests/kuery/test_dsl.py::TestKQLtoDSL::test_or_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ 53%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_and_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 54%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_boolean_precedence PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    [ 55%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_field_equals PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 56%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_field_inequality PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [ 56%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_ip_checks PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 57%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_list_of_values PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        [ 58%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_not_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 59%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_or_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 60%]
tests/kuery/test_eql2kql.py::TestEql2Kql::test_wildcard_field PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        [ 61%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_and_expr PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         [ 62%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_and_values PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 62%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_cidr_match PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 63%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_field_exists PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 64%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_flattening PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 65%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_list_value PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 66%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_not_value PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        [ 67%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_or_expr PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 68%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_or_values PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        [ 68%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_quoted_wildcard PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  [ 69%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_range PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 70%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_single_value PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 71%]
tests/kuery/test_evaluator.py::EvaluatorTests::test_wildcard PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         [ 72%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_and_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 73%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_boolean_precedence PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    [ 74%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_field_equals PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 75%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_field_inequality PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [ 75%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_list_of_values PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        [ 76%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_lone_value PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 77%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_nested_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 78%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_not_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 79%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_or_query PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 80%]
tests/kuery/test_kql2eql.py::TestKql2Eql::test_schema PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 81%]
tests/kuery/test_lint.py::LintTests::test_and_not PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    [ 81%]
tests/kuery/test_lint.py::LintTests::test_compound PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [ 82%]
tests/kuery/test_lint.py::LintTests::test_double_negate PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 83%]
tests/kuery/test_lint.py::LintTests::test_extract_not PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 84%]
tests/kuery/test_lint.py::LintTests::test_ip PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         [ 85%]
tests/kuery/test_lint.py::LintTests::test_lint_field PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ 86%]
tests/kuery/test_lint.py::LintTests::test_lint_precedence PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 87%]
tests/kuery/test_lint.py::LintTests::test_merge_fields PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               [ 87%]
tests/kuery/test_lint.py::LintTests::test_mixed_demorgans PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 88%]
tests/kuery/test_lint.py::LintTests::test_not_demorgans PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 89%]
tests/kuery/test_lint.py::LintTests::test_not_or PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     [ 90%]
tests/kuery/test_lint.py::LintTests::test_upper_tokens PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               [ 91%]
tests/kuery/test_parser.py::ParserTests::test_conversion PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 92%]
tests/kuery/test_parser.py::ParserTests::test_date PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [ 93%]
tests/kuery/test_parser.py::ParserTests::test_keyword PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 93%]
tests/kuery/test_parser.py::ParserTests::test_list_equals PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 94%]
tests/kuery/test_parser.py::ParserTests::test_multiple_types_fail PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    [ 95%]
tests/kuery/test_parser.py::ParserTests::test_multiple_types_success PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ 96%]
tests/kuery/test_parser.py::ParserTests::test_number_exists PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [ 97%]
tests/kuery/test_parser.py::ParserTests::test_number_wildcard_fail PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   [ 98%]
tests/kuery/test_parser.py::ParserTests::test_type_family_fail PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 99%]
tests/kuery/test_parser.py::ParserTests::test_type_family_success PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    [100%]

======================================================================================================================================================================================================================================================================== warnings summary =========================================================================================================================================================================================================================================================================
env/detection-rules-build/lib/python3.10/site-packages/_pytest/config/__init__.py:1171
  /Users/stryker/workspace/Elastic/detection-rules/env/detection-rules-build/lib/python3.10/site-packages/_pytest/config/__init__.py:1171: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: typeguard
    self._mark_plugins_for_rewrite(hook)

tests/test_all_rules.py: 199 warnings
  /Users/stryker/workspace/Elastic/detection-rules/tests/test_all_rules.py:691: DeprecationWarning: Please use assertTrue instead.
    self.assert_(rule.contents.data.note, f'{self.rule_str(rule)} note required for config information')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================================================================================================================================================================================================== 116 passed, 200 warnings in 14.85s ================================================================================================================================================================================================================================================================
(detection_dev) ➜  detection-rules git:(create-setup-cfg) ✗

```

</details>